execution/src/windows/execution.cc

```diff
@@ -117,11 +118,11 @@ namespace leatherman { namespace execution {
 
         // Format a name for the pipe based on the process and counter
         wstring name = boost::nowide::widen((boost::format("\\\\.\\Pipe\\leatherman.%1%.%2%") %
-            GetCurrentProcessId() %
+            GetCurrentThreadId() %
```

Comment:
We use NamedPipe specifically for OVERLAPPED I/O. In this case we're also creating both sides of the pipe.

I'm not sure I follow your reasoning why we also need process ID, as thread ID should be unique system-wide.